packages/medical-domain/src/domains/repositories/medical/MedicalDataService.ts

```diff
@@ -138,6 +181,11 @@ class MedicalDataService {
   }
 
   private join(): void {
+    this.joinBolus()
```

Comment:
I don't know any of this code, can you explain it to me?

Reply:
Yes 😄 This intends to fix a bug in the `ReservoirChange` object, where the `ReservoirChange.pump` property was not filled.

When the medical data is received, it is processed through the `add()` method (see line 418). Each object type of the medical data is formatted in the `normalize()` method, and then, if needed, some jointures between 2 objects can be done in the `join()` method. It was already the case for the `bolus` data, so I added another method to enrich each `ReservoirChange` object with the data from the `PumpSettings`.

In order to fill the `ReservoirChange.pump` object, we want to take the most recent `PumpSettings`, and retrieve only the `PumpSettings.pumpConfig` property. 

That's my understanding of the subject but I may have misunderstood some things 👼